ansible_base/serializers/resource.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import re
+import uuid
+
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+from rest_framework import serializers
+from rest_framework.reverse import reverse_lazy
+
+from ansible_base.models import Resource, ResourceType
+
+logger = logging.getLogger('ansible_base.serializers')
+
+ANSIBLE_ID_REGEX = re.compile(r"^[0-9a-fA-F]{8}:[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")
+
+
+def get_resource_detail_view(resource: Resource):
+    # TODO: format this so that it uses the correct API base path for the gateway.
+    actions = resource.content_type.resource_type.get_resource_config().get("actions")
+
+    # TODO: this needs some more logic to handle cases where the detail view isn't
+    # name or pk, and in cases where there may be multiple detail views (such as with
+    # nested API views). This may be solvable by providing a reverse_url_name when
+    # resources are registered.
+    if detail := actions.get("retrieve"):
+        return detail[0][1].format(pk=resource.object_id, name=resource.name)
+
+    return None
+
+
+def ansible_id_validator(val):
+    message = _(
+        "Ansible ID must in the format of <service_id>:<resource_id>, "
+        "where service_id is the first 8 characters of the service's ID "
+        "and resource_id is a valid UUID 4."
+    )
+
+    if not ANSIBLE_ID_REGEX.match(val):
+        raise serializers.ValidationError(message)
+
+    (s_id, r_id) = val.split(":")
+
+    try:
+        if str(uuid.UUID(r_id, version=4)) != r_id:
+            raise serializers.ValidationError(message)
+    except ValueError:
+        raise serializers.ValidationError(message)
```

Comment:
Can this function be simplified?
```
def ansible_id_validator(val):
    matches = re.match("^[0-9a-f]{8}:(.*)$", val, re.IGNORECASE)
    try:
        if not matches:
            raise ValueError
        print("Checking UUID")
        print(matches.group(1))
        if str(uuid.UUID(matches.groups(1), version=4)) != matches.groups(1):
            raise ValueError
    except (AttributeError, ValueError):
        raise serializers.ValidationError(_(
            "Ansible ID must in the format of <service_id>:<resource_id>, "
            "where service_id is the first 8 characters of the service's ID "
            "and resource_id is a valid UUID 4."
        ))
```

Reply:
Can this also be moved into utils/validation.py or somewhere else besides a serializer?